docs/vision.md

```diff
@@ -1,4 +1,24 @@
+
+But bloqade is not done with just these two components. We envision adding new components (called "dialects") which help you write programs which are tuned for optimal performance in an error corrected era, and on neutral atom hardware. Stay tuned and help us build the future of quantum computing as we build out new components, such as QEC and atom moving dialects.
+
+
+# Hardware-oriented programming and co-design
+At its core, Bloqade strives to be the neutral atom SDK for getting the most out of today's and tomorrows' quantum hardware. It is clear that the circuit-level abstraction is not enough to program real quantum hardware; indeed, tomorrows' quantum demonstrations and applications must program at the hardware level and develop special tooling to compile higher-level abstractions to efficient implementations. We call this process **"co-design"**: designing algorithms specialized to near-term hardware, with an eye on nontrivial demonstrations and scalable solutions. Ultimately, this co-design approach requires hardware-specific DSLs which explicitly represent the native executions on neutral atom hardware: in other words, Bloqade.
+
+
+# Hybrid computing beyond circuits
+Many quantum algorithms are hybrid, requiring both classical and quantum resources to work together in a hybrid computation architecture. This could be anything from syndrome extraction and measurement-based computing to variational parameter updates in VQE methods and orbital fragmentation methods in molecular simulation. Through the use of the Kirin compiler infrastructure, Bloqade embraces this philosophy of heterogeneous compute. Kirin programs are written as (compositions of) [kernels](https://en.wikipedia.org/wiki/Compute_kernel)-- subroutines that are intended to run on particular hardware (such as QPUs), or orchestrated to run on heterogeneous compute (such as a real-time classical runtime plus a QPU). These subroutines-- plus the built-in hybrid representations-- enable many key primitives, such as error correction.
+
+Additionally, the ability to compose functions together and to use typical classical programming structures like `if` and `while` enables many simplifications in writing raw circuit executions. In fact, `while` loops and the ability to dynamically allocate new memory (which is not known until runtime) enables many powerful subroutines and is natively enabled with bloqade's kernel-based representation; for example, see [this implementation](digital/examples/repeat_until_success.py) of a repeat-until-success program.
```

Comment:
I thought about this paragraph a bit more. Maybe we should also mention that control flows enable creating abstraction? such as recursion, functions, dispatching based on given options etc. Even they are eventually flattened. I'm not sure what would be the best to insert this into this paragraph.